app/lib/nokul/version.rb

```diff
@@ -1,5 +1,24 @@
 # frozen_string_literal: true
 
 module Nokul
-  VERSION = '0.5.8'
+  class Version
```

Comment:
Versiyon parse + representation için tamamen custom bir class yazmak yerine bence standart dağıtımla gelen [Gem::Version](https://docs.ruby-lang.org/en/2.6.0/Gem/Version.html) kullanalım (Ruby gem'leri de bu sınıfı kullanıyor).  Örnek:

```ruby
version = Gem::Version.new '0.5.8'
version.canonical_segments #=> ['0,' ,'5', '8']

other = Gem::Version.new '0.6'
other > version #=> true
```

Buna dayalı olarak bu sınıf şu hali alabilir:

```ruby
module Nokul
  class Version < Gem::Version
    def self.load_from_manifest
      version = if File.exist?(manifest = Rails.root.join('app.json'))
                  JSON.parse(File.read(manifest)).fetch('version')
                end
      new(version || '0.0.0')
    end

    def self.current
      @current ||= load_from_manifest.to_s
    end
  end
end
```

(Belki bir sonraki refaktörde manifest'ten sürüm yükleme lojiğini Nokul::Support'a almayı da düşünebiliriz.  Nokul'a özgü bir hali yok bu lojiğin).